js/classes/entities/foreground/weather_system.js

```diff
@@ -0,0 +1,182 @@
+/**
+ * WeatherSystem manages what weather to spawn into a zone.
+ * 
+ * @author Caleb Krauter
+ */
+class WeatherSystem {
+
+    /**
+     * Pass in the type of weather, intensity and time of day.
+     * Direction of rain or snow is random.
+     * Rain or snow will always include clouds.
+     * Intensity is used for both rain/snow and clouds.
+     * 
+     * @param {String} type warm, hot, cloudy, rain, snow.
+     * @param {int} intensity a value between 0 and 5.
+     * @param {String} time day or night.
+     */
+    constructor(type, intensity, time) {
+        this.type = type;
+        this.intensity = intensity;
+        this.time = time;
+
+        // Rainy day or night
+        this.makeHeavens();
+
+        if (this.type === "rain") {
+            this.makeClouds();
+            this.makePrecipitation();
+        }
+        else if (this.type === "snow") {
+            this.makeClouds();
+            this.makePrecipitation();
+            this.addSurfaceSnow();
+        } else if (this.type === "cloudy") {
+            this.makeClouds();
+            // Default to sunny with white clouds
+        }
+        // Defaults to sunny/(night with moon) and no clouds.
+        else {
+            this.makeHeavens();
+        }
+    }
+
+    /**
+     * Adds a sun/moon and updates the sky color.
+     */
+    makeHeavens() {
+        if (this.time === "night") {
+            BG_COLOR = COLORS.SKY_DARK_GREY;
+            GAME.addEntity(new Sun(new Vector(Camera.SIZE.x - 2 * Sun.SCALED_SIZE, Sun.SCALED_SIZE - 100), Sun.MOON), -1);
```

Comment:
Looks like this vector, 

```js
new Vector(Camera.SIZE.x - 2 * Sun.SCALED_SIZE, Sun.SCALED_SIZE - 100),
```

is used a number of times in this function. What do you think about defining it once as a variable somewhere earlier in the function, then just use the variable everywhere it is needed?

```js
const sunVector = new Vector(Camera.SIZE.x - 2 * Sun.SCALED_SIZE, Sun.SCALED_SIZE - 100)
```

and then this becomes, 

```suggestion
            GAME.addEntity(new Sun(sunVector, Sun.MOON), -1);
```

Reply:
Good idea!